GoogleCloudPlatform/emblem
# feat(delivery): moving cloud build triggers to terraform configs (part 1)

Related to task: #261 

**Triggers list:** https://console.cloud.google.com/cloud-build/triggers?project=emblem-ops

### Triggers included:
* `api-unit-tests`
* `api-push-to-main`
* `website-unit-tests`
* `web-push-to-main`

### Steps to test PR:
Access to [Terraform](go/terraform-codelab) is required to be locally installed to test the changes.
Optional (step 0?): Safe step to rename the resources differently than the manual triggers they will be replacing to make sure they are functioning as expected 

1. Load the PR locally and run the following
`terraform init && terraform validate`
`terraform plan`
`terraform apply`

2. Ensure that the triggers depicted in the terraform configs are applied correctly in the list of [triggers](https://pantheon.corp.google.com/cloud-build/triggers?project=emblem-ops) in emblem project and they match the manual triggers they are replacing

3. Create a dummy PR request that modifies a file within `content-api/` directory. When the PR is created, you should see the pre-commit step within Github and in the Cloud Build history [tab ](https://pantheon.corp.google.com/cloud-build/builds?project=emblem-ops) that the `api-unit-tests` trigger (created by terraform) is running the steps within  `ops/unit-tests.cloudbuild.yaml` as expected.

4. In the same PR, modify a file within `website/` directory and confirm that `website-unit-tests` trigger is executed.

5. Merge the above PR. This should run both `api-push-to-main` and `web-push-to-main` triggers. Verify that in the pre-commit steps, as well as within the project history [tab ](https://pantheon.corp.google.com/cloud-build/builds?project=emblem-ops)
